r5jxrm/r5jxrm/Program.cs

```diff
@@ -0,0 +1,1111 @@
+            tet = tet * 2;
+            return jatekosKor;
+        }
+
+        private static double MegadásMethod(double tet, out bool vesztett, out bool isPlaying)
+        {
+            tet = tet / 2;
+            Console.WriteLine("Megadtad! Csak a felét veszted el, a tétnek");
+            tovabbLepesSzovegClear();
+            vesztett = true;
+            isPlaying = false;
+            return tet;
+        }
+
+        private static void SzetvalasztasMethod(ref double tet, double ennyivan, ref bool errorSave, ref bool splitting, ref bool hasSplit, ref int jatekosKartyaErtek, ref int jatekosSzetvalasztottKartyaErtek, string[] jatekosKartyai, string[] jatekosSzetvalasztottKartyak, ref int i)
+        {
+            if (ennyivan - tet * 2 >= 0)
+            {
+                splitting = true;
+                hasSplit = true;
+                if (jatekosKartyai[1] == "A - One" && jatekosKartyai[0] == "A")
+                {
+                    jatekosKartyai[1] = "A";
+                    jatekosKartyaErtek += 10;
+                }
+                jatekosSzetvalasztottKartyak[0] = jatekosKartyai[1];
+                jatekosKartyaErtek -= convertToInt(jatekosKartyai, 1);
+                jatekosSzetvalasztottKartyaErtek += convertToInt(jatekosKartyai, 1);
+                jatekosKartyai[1] = "";
+                tet = tet * 2;
+                i = 0;
+                Console.Clear();
+            }
+            else
+            {
+                Console.WriteLine("Nincs elég zsetonod szétválasztani!");
+                errorSave = true;
+                tovabbLepesSzovegClear();
+            }
+        }
+
+        private static double VegeredmenyHaNemValasztottSzet(ref double tet, bool vesztett, bool nyert, double biztositottZseton, int osztoKartyaErtek, int jatekosKartyaErtek)
+        {
+            if (vesztett == false && osztoKartyaErtek < jatekosKartyaErtek || nyert == true && vesztett == false)
+            {
+                Console.WriteLine($"Nyertél! Nyereményed: {tet}");
+                return (tet - biztositottZseton);
+            }
+            else if (osztoKartyaErtek == jatekosKartyaErtek || nyert == true && vesztett == true)
+            {
+                Console.WriteLine("Döntetlen! Nincs zseton nyereség vagy veszteség.");
+                tet = 0 - biztositottZseton;
+                return tet;
+            }
+            else if (vesztett == true && nyert == false || osztoKartyaErtek > jatekosKartyaErtek)
+            {
+                Console.WriteLine($"Vesztettél! A veszteség: {tet}");
+                tet = -tet - biztositottZseton;
+                return tet;
+            }
+            else
+            {
+                return tet;
+            }
+        }
+
+        private static double VegeredmenyHaSzetvalasztott(ref double tet, bool splitLost, bool vesztett, bool nyert, double biztositottZseton, int osztoKartyaErtek, int jatekosKartyaErtek, int jatekosSzetvalasztottKartyaErtek)
```

Comment:
Ha ennyi paramétere van egy metódusnak, akkor érdemes egy osztályba becsomagolni a paramétereket, mert  a hívás helyéről nehéz kitalálni, hogy hogyan is kéne viselkednie a metódusnak vagy hogy mi várható, mi történjen.